app/models/casebook/contents.rb

```diff
@@ -0,0 +1,102 @@
+# == Schema Information
+#
+# Table name: casebooks
+#
+#  id            :integer          not null, primary key
+#  title         :string           default("Untitled casebook"), not null
+#  slug          :string
+#  subtitle      :string
+#  headnote      :text
+#  public        :boolean          default(TRUE), not null
+#  root_id       :integer
+#  ordinals      :integer          default([]), not null, is an Array
+#  copy_of_id    :integer
+#  is_alias      :boolean
+#  material_type :string
+#  material_id   :integer
+#  created_at    :datetime         not null
+#  updated_at    :datetime         not null
+#
+
+class Casebook::Contents < Casebook::Generic
+  default_scope {where.not(root_id: nil)}
```

Comment:
`root_id` should probably  be `book_id`. Every piece of content that isn't a book stores the id of the book it belongs to. So `root_id` + `ordinals` uniquely identifies every piece of content in the database.

Each piece of content could also store its immediate parent, but that's not necessary to represent the tree.